core/src/main/scala/org/apache/spark/executor/TaskMetrics.scala

```diff
@@ -39,42 +39,67 @@ class TaskMetrics extends Serializable {
   /**
    * Host's name the task runs on
    */
-  var hostname: String = _
-
+  private var _hostname: String = _
+  def hostname = _hostname
+  def setHostname(value : String) = _hostname = value
+  
   /**
    * Time taken on the executor to deserialize this task
    */
-  var executorDeserializeTime: Long = _
-
+  private var _executorDeserializeTime: Long = _
+  def executorDeserializeTime = _executorDeserializeTime
+  def incExecutorDeserializeTime(value: Long) = _executorDeserializeTime += value
```

Comment:
can we make all of the increment/decrement functions private[spark]? I forgot to write that in the JIRA, but basically the main point of this patch is to make those immutable for users outside of Spark.